ui/src/main/webapp/src/app/shared/validators/project-name-not-exists.validator.ts

```diff
@@ -25,7 +29,10 @@ export class ProjectNameNotExistsValidator implements Validator {
                 if (result == null) {
                     resolve(null);
                 } else {
-                    resolve({['nameIsTaken']: true});
+                    if (this.project != null && result == this.project.id)
```

Comment:
That is possible solution too. I was also thinking about not sending the request to server at all - maybe only compare current project name with user input?

But your solution is probably more robust